### PR TITLE
Issue 25: Upgrade flink runtime to flink-1.15.2

### LINKED ISSUE #14: Use PravegaKeycloakCredentials class
Replace https://github.com/pravega/flink-tools/blob/master/flink-tools/src/main/java/io/pravega/flinktools/util/PravegaKeycloakCredentialsFromString.java with https://github.com/pravega/pravega-keycloak/blob/master/client/src/main/java/io/pravega/keycloak/client/PravegaKeycloakCredentials.java

### LINKED ISSUE #25: Upgrade flink runtime to 1.15.2
- Current version of flink(1.10.2) supported by flink-tools is very old as compared to the latest flink (1.15)
- Update/Remove deprecated SDP object resources
- Update helm stable repo

### PR DESCRIPTION
**Pull request by rishisuresh7**
# Upgrade flink runtime to 1.15.2
- Fixes issues
  - Closes #25
  - Fixes #14 
- Handle flink API deprecations and dependencies
  - upgrade gradle dependencies to support flink-1.15.2
  - wrap flink type `Row` to a comparable type `ComparableRow` (Due to the conversion of flink `Row` type to a final class, this type cannot be extended anymore. A wrapper class for the same has been implemented).
  - implement a custom serializer `AvroSchemaSerialzer` for reading/writing avro `GenericRecords` to process parquet files with flink
- Remove parsing of keycloak string to load keycloak credentials via ENV.
  - Use basic auth when the parameters `username` and `password` are provided to the jar.
- Remove local flink-parquet and use a released version
  - remove deprecated `ParquetAvroWriter` and introduce `AvroParquetWriter`
- Update flink-tools chart to latest compatible version in SDP
  - remove deprecated field `mavenCoordinate` and add fields `language`, `mainApplicationType` and `mainApplicationFile` 
- Update build scripts
  - add latest stable helm chart repository
  - update `kubectl` commands for better processing
- Update REAMDE

- Running apps in SDP with flink-1.15.2
   samples
- Clusters used for testing
  clusters
- Parquet files being generated by the job 
  parquet-files
- Verification of content in generated parquet files
  parquet-files-content

### CODE REVIEW

File: flink-tools/src/main/java/io/pravega/flinktools/StreamToParquetFileJob.java [hunk 1]
```diff
@@ -116,10 +120,8 @@ public void run() {
                 toOutput.print("output");
             }
 
-            // You need to overwrite the flink-parquet as a shadowJar along with withCompressionCodec method calling
-            // when plan to use CompressionCodecName other than UNCOMPRESSED
             final StreamingFileSink<GenericRecord> sink = StreamingFileSink
-                    .forBulkFormat(new Path(outputFilePath), ParquetAvroWriters.forGenericRecord(outputSchema).withCompressionCodec(CompressionCodecName.SNAPPY))
```
Comment (vangork) [inline flink-tools/src/main/java/io/pravega/flinktools/StreamToParquetFileJob.java:122]: Can you clarify how you reimlement the compression after  `.withCompressionCodec(CompressionCodecName.SNAPPY))` is deleted?

Comment (rishisuresh7) [reply flink-tools/src/main/java/io/pravega/flinktools/StreamToParquetFileJob.java:122]: Sorry I missed the compression part, I have added the writers for supporting different compression methods.
- Running jobs
image
- Compressed parquet files
image

**Comment by vangork** [comment]
LGTM now.

**Review by vangork — APPROVED** [review]
LGTM now.

**Review by ShwethaSNayak — APPROVED** [review]
LGTM

**Comment by RaulGracia** [comment]
@rishisuresh7 please, create an issue to describe the problem you are fixing in this PR before we merge. PRs should always be associated with a Git issue.

**Comment by rishisuresh7** [comment]
@RaulGracia , thanks for the review! I have created an issue and updated the PR with the same.

**Pull request merged**